Tests/CacheAdvanceTests/CacheHeaderHandleTests.swift

```diff
@@ -101,84 +97,114 @@ final class CacheHeaderHandleTests: XCTestCase {
-            offsetInFileAtEndOfNewestMessage: 500)
+            version: 1)
+        try originalHeader.synchronizeHeaderData()
+
+        let sut = try createHeaderHandle(
+            maximumBytes: 1000,
+            overwritesOldMessages: true,
+            version: 2)
+
+        XCTAssertFalse(try sut.canOpenFile())
+    }
 
+    func test_canOpenFile_emptyFile_throwsFileCorrupted() throws {
         let sut = try createHeaderHandle(
             maximumBytes: 1000,
             overwritesOldMessages: true,
             version: 2)
 
-        XCTAssertFalse(sut.canOpenFile(with: fileHeader))
+        XCTAssertThrowsError(try sut.canOpenFile()) {
+            XCTAssertEqual($0 as? CacheAdvanceError, CacheAdvanceError.fileCorrupted)
+        }
     }
 
-    func test_canOpenFile_maximumBytesMismatch_returnsFalse() throws {
-        let fileHeader = FileHeader(
-            version: 1,
+    func test_canWriteToFile_maximumBytesMismatch_returnsFalse() throws {
+        let originalHeader = try createHeaderHandle(
             maximumBytes: 1000,
             overwritesOldMessages: true,
-            offsetInFileOfOldestMessage: FileHeader.expectedEndOfHeaderInFile,
-            offsetInFileAtEndOfNewestMessage: 500)
+            version: 1)
+        try originalHeader.synchronizeHeaderData()
 
         let sut = try createHeaderHandle(
             maximumBytes: 2000,
             overwritesOldMessages: true,
             version: 1)
 
-        XCTAssertFalse(sut.canOpenFile(with: fileHeader))
+        XCTAssertFalse(try sut.canWriteToFile())
     }
 
-    func test_canOpenFile_overwritesOldMessagesMismatch_returnsFalse() throws {
-        let fileHeader = FileHeader(
-            version: 1,
+    func test_canWriteToFile_overwritesOldMessagesMismatch_returnsFalse() throws {
+        let originalHeader = try createHeaderHandle(
             maximumBytes: 1000,
             overwritesOldMessages: true,
-            offsetInFileOfOldestMessage: FileHeader.expectedEndOfHeaderInFile,
-            offsetInFileAtEndOfNewestMessage: 500)
+            version: 1)
+        try originalHeader.synchronizeHeaderData()
 
         let sut = try createHeaderHandle(
             maximumBytes: 1000,
             overwritesOldMessages: false,
             version: 1)
 
-        XCTAssertFalse(sut.canOpenFile(with: fileHeader))
+        XCTAssertFalse(try sut.canWriteToFile())
+    }
+
+    func test_canWriteToFile_versionMismatch_returnsFalse() throws {
+        let originalHeader = try createHeaderHandle(
+            maximumBytes: 1000,
+            overwritesOldMessages: true,
+            version: 1)
+        try originalHeader.synchronizeHeaderData()
+
+        let sut = try createHeaderHandle(
+            maximumBytes: 1000,
+            overwritesOldMessages: true,
+            version: 2)
+
+        XCTAssertFalse(try sut.canWriteToFile())
+    }
+
+    func test_canWriteToFile_emptyFile_returnsFalse() throws {
```

Comment:
Can we add a test for when `canWriteToFile()` throws?

Reply:
when I next push, this will only throw on a `FileHandle` error, which is not something I can simulate.